### ISSUE TITLE
Datatype detection needs to take into account control flow

### ISSUE THREAD

**Issue by cfabianski** [Bearer/bearer-rules#127]
## Description & Reproduction

When we use the datatype detection, we should skip ternary expression 

```java
            logger.error(user.name ? "ok": "ok");
```

## Expected Behavior

No alerts gets triggered

## Actual Behavior

Leak of information in logger gets triggered

## Your Environment

- Operating System and version:
- Output of 'bearer version':
```
bearer version: 0.00.0
sha: xxx
```

**Comment by gotbadger** [comment]
fixed
